### ISSUE TITLE
Crash on reading title with foreign characters

### ISSUE THREAD

**Issue by 1bod** [therealmarius/Spotify-2-AppleMusic#15]
```sh
N°728 | にんじゃりばんばん - Steve Aoki Remix | Kyary Pamyu Pamyu, Steve Aoki | にんじゃりばんばん (Steve Aoki Remix) => NOT FOUND
Traceback (most recent call last):
  File "C:\Users\mrayn\OneDrive\Documents\GitHub\Spotify-2-AppleMusic\convertsongs.py", line 153, in <module>
    f.write(f'{title} | {artist} | {album} => NOT FOUND')
  File "C:\Python312\Lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'charmap' codec can't encode characters in position 0-8: character maps to <undefined>```

**Comment by nf1973** [comment]
The crash looks to be only with writing to the "_noresult" file.
Can you upload your playlist file? I tried to reproduce by creating a playlist file from your output, but it did not crash.

**Comment by 1bod** [comment]
It's quite big but here it is:
[liked.csv](https://github.com/therealmarius/Spotify-2-AppleMusic/files/13401250/liked.csv)

**Comment by nf1973** [comment]
For speed, I copied just that track where you have the crash, plus the one before and the one after into another file and still no crash (on a Mac).

```
N°1 | The Tempo | Bleu Clair | The Tempo => 1551405690
Song 1551405690 added to playlist test!
N°2 | にんじゃりばんばん - Steve Aoki Remix | Kyary Pamyu Pamyu, Steve Aoki | にんじゃりばんばん (Steve Aoki Remix) => NOT FOUND

N°3 | Fever - Feder Remix | Dua Lipa, Angèle, Feder | Fever (Feder Remix) => 1545657877
Song 1545657877 added to playlist test!

 - STAT REPORT -
Playlist Songs: 3
Converted Songs: 2
Failed Songs: 1
Playlist converted at 67%
```

One idea.. what happens if you change line 152 (of convertsongs.py)?
 from
`with open(f'{playlist_name}_noresult.txt', 'a+') as f:`

to
`with open(f'{playlist_name}_noresult.txt', 'a+', encoding='utf-8') as f:`

**Comment by 1bod** [comment]
I tried again with a shorter playlist it still crashed but after changing line 152 to the new one, it worked!
```
N°1 | The Tempo | Bleu Clair | The Tempo => 1551405690
Song 1551405690 added to playlist spotify!
N°2 | にんじゃりばんばん - Steve Aoki Remix | Kyary Pamyu Pamyu, Steve Aoki | にんじゃりばんばん (Steve Aoki Remix) => NOT FOUND
```
Thank you very much

**Comment by therealmarius** [comment]
Hi,
It seems that the issue is resolved and the code will be fixed with the #14 PR. I'll close this issue when it will be merged.
Thank you and enjoy your converted playlists!
